server/poll/poll.go

```diff
@@ -142,10 +143,20 @@ func (p *Poll) GetVotedAnswer(userID string) (*VotedAnswerResponse, error) {
 			}
 		}
 	}
-	return &VotedAnswerResponse{
-		PollID:       p.ID,
-		UserID:       userID,
-		VotedAnswers: votedAnswer,
+	return votedAnswer, nil
+}
+
+// GetMetadata returns personalized metadata of a poll
+func (p *Poll) GetMetadata(userID string, permission bool) (*Metadata, error) {
```

Comment:
Thank you for interesting suggestion. But I think that pointer type is fine here. Because...

* Return value of `GetMetadata` is used just as the receiver of funcs ([`metadata.EncodeToByte()`](https://github.com/matterpoll/matterpoll/pull/234/files#diff-4444f0caa92e553b5c018c908e90e5e6R541) and [`metadata.ToMap()`](https://github.com/matterpoll/matterpoll/pull/234/files#diff-4444f0caa92e553b5c018c908e90e5e6R234)). So if changing return type of `GetMetadata` to value type, we should change receiver of these funcs. But, as far as I know, the receiver of these funcs should be pointer type.
* If return type of `GetMetadata` is value type, we must return [the empty instance of `Metadata`](https://github.com/matterpoll/matterpoll/pull/234/files#diff-00ec2aff510fa90953e4a6d689e108fdR153) when error occurs in `GetMetadata`. It looks little strange.

I know that returning the value type might be better than returning the pointer type ([ref1](https://medium.com/@philpearl/bad-go-pointer-returns-340f2da8289), [ref2](https://stackoverflow.com/questions/23542989/pointers-vs-values-in-parameters-and-return-values/23551970#23551970)), but I think that pointer type is fine for the above reasons. 1/5

Of course, I haven't looked up everything about pointer vs value type, so I'm welcoming any references about this problem.